jplag/src/main/java/de/jplag/Submission.java

```diff
@@ -154,6 +154,8 @@ public boolean hasErrors() {
     public boolean parse(boolean debugParser) {
         if (files == null || files.size() == 0) {
             errorCollector.print("ERROR: nothing to parse for submission \"" + name, null);
+            tokenList = null;
```

Comment:
What is the benefit of setting the token list to null here?

Reply:
Not sure, but other code there does it too.

https://github.com/jplag/JPlag/blob/master/jplag/src/main/java/de/jplag/Submission.java#L167

https://github.com/jplag/JPlag/blob/master/jplag/src/main/java/de/jplag/Submission.java#L174

Might be an old convention though, code is changing rapidly.

What I wondered about is why does the comparison code check for `tokenList != null` when we are supposedly retrieving valid submissions from the submission set. It is likely also an old convention though.

It might very well be the case that the `tokenList` check was the original way to denote invalid, and `error` was added at a later date for some reason.

Reply:
I suspected that! I think in the long run the null checks in the strategies should probably be assertions or we should remove them and the submissions set should assert that all valid submission sets have non-null token lists. Then we can also remove setting null here. For now, let's keep it consistent, as you did here with setting it to null.

I will make an issue for that null-checks, null-setting, and error duplication!

Reply:
Documented in #249!